src/app/services/article.service.ts

```diff
@@ -28,106 +53,21 @@ export class ArticleService {
             return fetchedArticle ?? getHardcodedArticle(articleId);
           } catch (error) {
             console.error(`Failed to fetch article with ID: ${articleId}`, error);
-            return null; // Return null if there's an error
+            return []; // Return empty if there's an error
           }
         })
       );
-  
+
       // Filter out any null values
       this.articles = fetchedArticles.filter((article): article is Article => article !== null);
-  
-      console.log('Articles loaded:', this.articles);
-    } catch (error) {
-      console.error("Error loading articles", error);
-    }
-  };  
-  // Call the function to load articles  
-  stripHtml(html: string): string {
-    const div = document.createElement('div');
-    div.innerHTML = html;
-    return div.textContent || div.innerText || '';
-  }
-  
-  getArticleList(): Observable<Article[]> {
-    // Use 'from' to convert the Promise returned by loadArticles to an Observable
-    return from(this.loadArticles()).pipe(
-      // Switch to returning this.articles if it's available after loading
-      switchMap(() => {
-        if (this.articles?.length) {
-          return of(this.articles); // Return the articles if already loaded
-        }
-  
-        // Fallback to hardcoded articles after delay if this.articles is empty
-        return of(getHardcodedArticles()).pipe(
-          delay(1500), // Simulate delay (e.g., loading time)
-          tap((articles: Article[]) => this.articles = articles) // Update this.articles
-        );
-      })
-    );
-  }
-  readOne(articleId: string): Observable<Article> {
-    return of(getHardcodedArticles().find((article: Article) => article.articleId === articleId) || {
-      header: 'invalid-article',
-      body: 'invalid-article',
-      imageURI: 'none',
-      subCategory: 'Article',
-      articleId: 'invalid-article'
-    });
-  }
-
-  saveArticleBody = async (article:Article) => {
```

Comment:
Although this is not used yet, it will be soon. Can we agree not to delete methods before we had a chance to talk about it?

Reply:
Sure. However, it’s worth considering that if a function or otherwise other code isn't being used, it may be best to keep it in a local branch rather than including it in a push, until it’s actually needed.

If we consistently keep committing code that is not utilized I can name a few reasons this will not be ideal:

- Unnecessary complexity to the code base.
- Other developers spend time trying to understand code that may never be used.
- Creating new features may use the code that is not necessarily ready.
- Being part of the Main branches code history from creation to potentially eventual removal.

Reply:
I think you're overthinking this. We don't 20 developers in our team. 

Just assume that the other person knows what he's doing and if  you can't make that assumption, just ask. Deleting the code without asking is worse than any of those line items.

Reply:
Agreed